Do not use SSLv23_method()->put_cipher_by_char to determine size of cipher
on v23 hello messages. 

This method if used with NULL arguments will return just the size of a cipher
on a SSLv2 message. In libreSSL library this method will return 2 which is 
wrong. The size of a cipher in an V2 hello message is always 3.

=== modified file 'src/ssl/bio.cc'
--- src/ssl/bio.cc	2015-09-01 09:25:57 +0000
+++ src/ssl/bio.cc	2015-09-22 15:27:36 +0000
@@ -1099,9 +1099,7 @@
 
     if (ciphersLen) {
         const SSL_METHOD *method = SSLv23_method();
-        int cs = method->put_cipher_by_char(NULL, NULL);
-        assert(cs > 0);
-        for (unsigned int i = 0; i < ciphersLen; i += cs) {
+        for (unsigned int i = 0; i < ciphersLen; i += 3) {
             // The v2 hello messages cipher has 3 bytes.
             // The v2 cipher has the first byte not null
             // Because we are going to sent only v3 message we

